lib/actions/os.coffee

```diff
@@ -134,50 +144,125 @@ exports.download =
+
+		Examples:
+
+			$ resin os build-config ../path/rpi3.img raspberrypi3 --output rpi3-config.json
+			$ resin os configure ../path/rpi3.img 7cf02a6 --config "$(cat rpi3-config.json)"
+	'''
+	permission: 'user'
+	options: [
+		{
+			signature: 'advanced'
+			description: 'show advanced commands'
+			boolean: true
+			alias: 'v'
+		}
+		{
+			signature: 'output'
+			description: 'the path to the output JSON file'
+			alias: 'o'
+			required: 'the output path is required'
+			parameter: 'output'
+		}
+	]
+	action: (params, options, done) ->
+		fs = require('fs')
+		Promise = require('bluebird')
+		writeFileAsync = Promise.promisify(fs.writeFile)
+
+		buildConfig(params.image, params['device-type'], options.advanced)
+		.then (answers) ->
+			writeFileAsync(options.output, JSON.stringify(answers, null, 4))
+		.nodeify(done)
+
 exports.configure =
 	signature: 'os configure <image> <uuid>'
 	description: 'configure an os image'
 	help: '''
-		Use this command to configure a previously download operating system image with a device.
+		Use this command to configure a previously downloaded operating system image for the specific device.
 
 		Examples:
 
 			$ resin os configure ../path/rpi.img 7cf02a6
 	'''
 	permission: 'user'
 	options: [
-		signature: 'advanced'
-		description: 'show advanced commands'
-		boolean: true
-		alias: 'v'
+		{
+			signature: 'advanced'
+			description: 'show advanced commands'
+			boolean: true
+			alias: 'v'
+		}
+		{
+			signature: 'config'
+			description: 'stringified JSON with the device config, see `resin os build-config`'
+			parameter: 'config'
+		}
 	]
 	action: (params, options, done) ->
-		_ = require('lodash')
 		resin = require('resin-sdk-preconfigured')
-		form = require('resin-cli-form')
 		init = require('resin-device-init')
 		helpers = require('../utils/helpers')
 
 		console.info('Configuring operating system image')
 		resin.models.device.get(params.uuid).then (device) ->
-			helpers.getManifest(params.image, device.device_type)
-			.get('options')
-			.then (questions) ->
-
-				if not options.advanced
-					advancedGroup = _.findWhere questions,
-						name: 'advanced'
-						isGroup: true
-
-					if advancedGroup?
-						override = helpers.getGroupDefaults(advancedGroup)
-
-				return form.run(questions, { override })
-			.then (answers) ->
-				init.configure(params.image, params.uuid, answers).then(helpers.osProgressHandler)
+			if options.config
+				return JSON.parse(options.config)
+			buildConfig(params.image, device.device_type, options.advanced)
+		.then (answers) ->
+			init.configure(params.image, params.uuid, answers).then(helpers.osProgressHandler)
 		.nodeify(done)
 
-initWarningMessage = '''
+exports.availableDrives =
+	# TODO: dedupe with https://github.com/resin-io-modules/resin-cli-visuals/blob/master/lib/widgets/drive/index.coffee
+	signature: 'os available-drives'
+	description: 'list available drives'
+	action: ->
+		Promise = require('bluebird')
```

Comment:
We use the local imports almost everywhere to speedup the runtime. Lodash is at the top because it's anyway imported in the top-level in some util file, and also because it's needed to build the option at start time.